# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gpm/open_max.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- gpm-1.20.1/src/prog/gpm-root.y.vanilla	2008-03-17 11:50:13.000000000 +0100
+++ gpm-1.20.1/src/prog/gpm-root.y	2008-03-17 11:50:54.000000000 +0100
@@ -44,7 +44,6 @@
 #include <sys/stat.h>       /* fstat() */
 #include <sys/utsname.h>    /* uname() */
 #include <termios.h>        /* winsize */
-#include <linux/limits.h>   /* OPEN_MAX */
 #include <linux/vt.h>       /* VT_ACTIVATE */
 #include <linux/keyboard.h> /* K_SHIFT */
 #include <utmp.h>         
@@ -525,7 +524,9 @@
 	            open("/dev/null",O_RDONLY); /* stdin  */
 	            open(consolename,O_WRONLY); /* stdout */
 	            dup(1);                     /* stderr */  
-	            for (i=3;i<OPEN_MAX; i++) close(i);
+                    int open_max = sysconf(_SC_OPEN_MAX);
+                    if (open_max == -1) open_max = 1024;
+                    for (i=3;i<open_max; i++) close(i);
 	            execl("/bin/sh","sh","-c",self->arg,(char *)NULL);
 	            exit(1); /* shouldn't happen */
 	         default: return 0;
--- gpm-1.20.1/src/special.c.vanilla	2008-03-17 11:45:35.000000000 +0100
+++ gpm-1.20.1/src/special.c	2008-03-17 11:51:32.000000000 +0100
@@ -25,7 +25,6 @@
 
 /* This file is compiled conditionally, see the Makefile */
 
-#include <linux/limits.h> /* for OPEN_MAX */
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -155,7 +154,9 @@
       open(GPM_NULL_DEV,O_RDONLY); /* stdin  */
       open(option.consolename,O_WRONLY); /* stdout */
       dup(1);                     /* stderr */
-      for (i=3;i<OPEN_MAX; i++) close(i);
+      int open_max = sysconf(_SC_OPEN_MAX);
+      if (open_max == -1) open_max = 1024;
+      for (i=3;i<open_max; i++) close(i);
       execl("/bin/sh","sh","-c",command,(char *)NULL);
       exit(1); /* shouldn't happen */
       
